storm/check.py

```diff
@@ -0,0 +1,573 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+
+# 3rd party
+import requests
+
+# project
+from checks import AgentCheck
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'storm'
+
+
+def _g(stat_map, default, func, *components):
+    """ Helper method to return value, tag tuple from a stat map get.
+
+    :param stat_map: stat map
+    :param default: default value
+    :param func: function to apply after getting the value.
+    :param components: components in order to traverse
+    :return: stat value
+    """
+    value = stat_map
+    for component in components:
+        if isinstance(component, (int, long)):
+            if isinstance(value, (list, tuple)) and len(value) >= component:
+                value = value[component]
+            else:
+                return default
+        else:
+            if component not in value:
+                return default
+            else:
+                value = value[component]
+    if value not in (None, ''):
+        if func is not None:
+            try:
+                return func(value)
+            except Exception:
+                return default
+        return value
+    else:
+        return default
+
+
+def _float(v):
+    """Try to convert to a float
+
+    :param v: value
+    :rtype: float
+    """
+    try:
+        return float(v)
+    except Exception:
+        return 0.0
+
+
+def _long(v):
+    """Try to convert to a long
+
+    :param v: value
+    :rtype: long
+    """
+    try:
+        return long(v)
+    except Exception:
+        return 0
+
+
+def _get_length(stat_map, default, *components):
+    """ Helper Function to get the length of an array type from the map.
+
+    :param stat_map: stat map
+    :param default: default length value
+    :param components: components in order to traverse.
+    :return: length of array or default value.
+    :rtype: int
+    """
+    return _g(stat_map, default, len, *components)
+
+
+def _get_long(stat_map, default, *components):
+    """ Helper Function to get the long value from the map.
+
+    :param stat_map: stat map
+    :param default: default length value
+    :param components: components in order to traverse.
+    :return: long of value or default value.
+    :rtype: long
+    """
+    return _g(stat_map, default, _long, *components)
+
+
+def _get_float(stat_map, default, *components):
+    """ Helper Function to get the float value from the map.
+
+    :param stat_map: stat map
+    :param default: default length value
+    :param components: components in order to traverse.
+    :return: float of value or default value.
+    :rtype: float
+    """
+    return _g(stat_map, default, _float, *components)
+
+
+class StormCheck(AgentCheck):
+    """
+    Apache Storm 1.x.x Topology Execution Stats
+    """
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
```

Comment:
I don't think it's necessary to initialized field here since we will override them in `update_from_config` at each `check` call. We could remove the `__init__` method I guess.